engine/src/snapshot_api/mod.rs

```diff
@@ -0,0 +1,80 @@
+pub use self::{memory_snapshot::MemorySnapshot, rocks_snapshot::RocksSnapshot};
+
+/// Memory snapshot
+mod memory_snapshot;
+/// Rocksdb snapshot
+mod rocks_snapshot;
+
+/// This trait is a abstraction of the snapshot, We can Read/Write the snapshot like a file.
+#[async_trait::async_trait]
+#[allow(clippy::pedantic)] // In order to unify the naming
```

Comment:
I tried remove this line, clippy doesn't warn or report errors.

Reply:
Removed